src/utils/recorder_events.jl

```diff
@@ -59,24 +59,62 @@ end
 
 function InitialConditionUpdateEvent(
     simulation_time,
-    key::ICKey,
     ic::InitialCondition,
-    val::Float64,
-    previous_value::Float64,
     problem_number::Int,
 )
+    value = get_value(ic)
+    if value isa PJ.ParameterRef
+        value = PJ.value(value)
+    end
+    previous_value = get_condition(ic)
+    if previous_value isa PJ.ParameterRef
+        previous_value = PJ.value(previous_value)
+    end
     return InitialConditionUpdateEvent(
         IS.RecorderEventCommon("InitialConditionUpdateEvent"),
         simulation_time,
-        string(get_entry_type(key)),
-        string(get_component_type(key)),
+        string(get_ic_type(ic)),
+        string(get_component_type(ic)),
         get_component_name(ic),
-        val,
+        PJ.value(get_value(ic)),
         previous_value,
         problem_number,
     )
 end
 
+struct ParameterUpdateEvent <: IS.AbstractRecorderEvent
+    common::IS.RecorderEventCommon
+    category::String
+    simulation_time::Dates.DateTime
+    parameter_type::String
+    device_name::String
+    previous_value::Float64
+    val::Float64
+    problem_number::Int
+    source::Int
+end
```

Comment:
This is not correct and TBD. Let me know what fields should be recorded for Parameter Updates. The implementation is as of now unclear.